support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.jsx

```diff
@@ -203,27 +195,31 @@ const StripeForm = (props: StripeFormPropTypes) => {
   };
 
   const handleCardErrors = () => {
-    // eslint-disable-next-line array-callback-return
-    ['cardNumber', 'cardExpiry', 'cardCvc'].map((field) => {
-      if (cardFieldsData[field].empty === true) {
-        setCardFieldsData(prevData => ({
-          ...prevData,
-          [field]: {
-            ...prevData[field],
-            error: cardFieldsData[field].errorEmpty,
-          },
-        }));
-      } else if (!cardFieldsData[field].complete) {
-        setCardFieldsData(prevData => ({
-          ...prevData,
-          [field]: {
-            ...prevData[field],
-            error: prevData[field].errorIncomplete,
-          },
-        }));
-      }
-      setCardErrors(getAllCardErrors());
-    });
+    const newCardFieldsErrors = [];
+    const newCardFieldsData = Object.entries(cardFieldsData)
+      .reduce((newData, [cardFieldName, existingFieldData]) => {
+        if (existingFieldData.empty) {
+          const error = existingFieldData.errorEmpty;
+          // eslint-disable-next-line no-param-reassign
+          newData[cardFieldName] = {
+            ...existingFieldData,
+            error,
+          };
+          newCardFieldsErrors.push({ field: [cardFieldName], message: error });
+        } else if (!existingFieldData.complete) {
+          const error = existingFieldData.errorIncomplete;
+          // eslint-disable-next-line no-param-reassign
+          newData[cardFieldName] = {
+            ...existingFieldData,
+            error,
+          };
+          newCardFieldsErrors.push({ field: [cardFieldName], message: error });
+        }
+        return newData;
+      }, { ...cardFieldsData });
```

Comment:
it looks like this whole thing is replacing the "error" in the field data based on errorEmpty and errorIncomplete result, and also building up `newCardFieldsErrors` a mapping from field name to error.
However it looks like it's editing in place the newData within the reduce, but then are we relying on that side effect as we then do a setCardFieldsData below with the same result.
I might have my scala hat on but I feel like it might be clearer use `map` or similar rather than `reduce` and also build a fresh data structure with the "data" and "errors".  Then that can be `unzip`ped or similar to get the two separate results.

Reply:
>  build a fresh data structure with the "data" and "errors"

I think this is a good suggestion, I think it could be clearer than modifying `newCardFieldsErrors` as a side effect of the reduce.